Automattic/wp-calypso
# Staging Sites: staging sync UI

Related to https://github.com/Automattic/dotcom-forge/issues/3956

## Proposed Changes

This pr adds a new UI to support staging sync.

## Testing Instructions

### Test 1: Staging side
1. Create a WoA site.
2. Add a staging site.
3. While on **the staging side**, ensure that you see the following UI:
![2023-10-05T14:05:20,185922676+03:00](https://github.com/Automattic/wp-calypso/assets/497103/99d3c848-4d80-4cb8-8bae-bc6fe946a7a0)
4. Select **Push to production**. Ensure that if no options are selected, the **Synchronize** button cannot be pressed. Check one ( or more ) item and press the **Synchronize** button. Ensure that you see the following:
![2023-10-03T15:43:19,808194242+03:00](https://github.com/Automattic/wp-calypso/assets/497103/bbcff18b-6f05-444a-b013-83ad603922dc)
5. Ensure that if the **production site slug** is not typed, the confirmation button will not be enabled.
6. Type your site slug, and press synchronize.
7. Ensure, using your dev console that a payload named `options` is passed as a parameter.
8. Choose **Pull from production**. Ensure that you see the following UI:
![2023-10-05T14:07:28,405097884+03:00](https://github.com/Automattic/wp-calypso/assets/497103/0aece32e-8f48-41ef-810f-6a5b1a280ffc)
9. Press the **Synchronize** button. Ensure that you see the following:
![2023-10-03T15:54:24,935317368+03:00](https://github.com/Automattic/wp-calypso/assets/497103/7b262b9c-6ed0-4057-925f-f1f5c77f21e3)
10. Press `Synchronize` and ensure that no options are passed as a payload.
11. Ensure that you see the following screen:
![2023-10-05T14:09:05,482668175+03:00](https://github.com/Automattic/wp-calypso/assets/497103/ab4b46b4-82a2-43e8-a713-798fcd1fa3a1)


### Test 1: Production side
1. Create a WoA site.
2. Add a staging site.
3. While on **the production side**, ensure that you see the following UI:
![2023-10-05T14:16:18,408199774+03:00](https://github.com/Automattic/wp-calypso/assets/497103/cfb0ffd6-afa6-47e7-b285-8f09fd74f252)
4. Select **Pull from staging**. Ensure that if no options are selected, the **Synchronize** button cannot be pressed. Check one ( or more ) item and press the **Synchronize** button. Ensure that you see the following:
![2023-10-03T15:43:19,808194242+03:00](https://github.com/Automattic/wp-calypso/assets/497103/bbcff18b-6f05-444a-b013-83ad603922dc)
5. Ensure that if the **production site slug** is not typed, the confirmation button will not be enabled.
6. Type your site slug, and press synchronize.
7. Ensure, using your dev console that a payload named `options` is passed as a parameter.
8. Choose **push to staging**. Ensure that you see the following UI:
![2023-10-05T14:18:17,425500451+03:00](https://github.com/Automattic/wp-calypso/assets/497103/1fff84bc-a018-4e2f-a2c8-232605b6621a)
9. Press the **Synchronize** button. Ensure that you see the following:
10. Press `Synchronize` and ensure that no options are passed as a payload. 
11. Ensure that you see the following screen:
![2023-10-05T14:19:09,816662979+03:00](https://github.com/Automattic/wp-calypso/assets/497103/80e5b80d-dc3b-4b0d-a4aa-156b9c642940)





## Pre-merge Checklist
